ribs/visualize.py

```diff
@@ -453,3 +453,145 @@ def sliding_boundaries_archive_heatmap(archive,
+    but the results should be interpreted carefully.
+
+    Examples:
+        .. plot::
+            :context: close-figs
+
+            >>> import numpy as np
+            >>> import matplotlib.pyplot as plt
+            >>> from ribs.archives import GridArchive
+            >>> from ribs.visualize import grid_archive_heatmap
+            >>> # Populate the archive with the negative sphere function.
+            >>> archive = GridArchive([20, 20, 20], [(-1, 1), (-1, 1), (-1, 1)])
+            >>> archive.initialize(solution_dim=2)
+            >>> for x in np.linspace(-1, 1, 100):
+            ...     for y in np.linspace(-1, 1, 100):
+            ...         for z in np.linspace(-1, 1, 100):
+            ...             archive.add(solution=np.array([x,y,z]),
+            ...                         objective_value=-(x**2 + y**2 + z**2),
+            ...                         behavior_values=np.array([x,y,z]))
+            >>> # Plot a heatmap of the archive.
+            >>> plt.figure(figsize=(8, 6))
+            >>> parallel_axes_plot(archive)
+            >>> plt.title("Negative sphere function")
+            >>> plt.ylabel("axis values")
+            >>> plt.show()
+
+    Args:
+        archive (SlidingBoundariesArchive): A 2D SlidingBoundariesArchive.
+        ax (matplotlib.axes.Axes): Axes on which to plot the heatmap. If None,
+            the current axis will be used.
+        bc_order (list<str>): By default, the BCs of the parallel axes will
+            appear in-order. To switch the axes around, supply a list of the
+            behavior axes (e.g., ['behavior_1', 'behavior_0', 'behavior_2']).
+        axis_labels(list<str>): By default, the BCs of the parallel axes will be
+            titled with the label 'behavior_x'. If the behavior dimensions have
+            a meaningful label, they can be supplied in order as the labels.
+        cmap (str, list, matplotlib.colors.Colormap): Colormap to use when
+            plotting intensity. Either the name of a colormap, a list of RGB or
+            RGBA colors (i.e. an Nx3 or Nx4 array), or a colormap object.
+        linewidth (float): Line width for solutions on the plot.
+        alpha (float): Opacity of solutions on graph (you may want to turn this
+            down if there are many solutions that are found to see them)
+        vmin (float): Minimum objective value to use in the plot. If None, the
+            minimum objective value in the archive is used.
+        vmax (float): Maximum objective value to use in the plot. If None, the
+            maximum objective value in the archive is used.
+
+    Raises:
+        ValueError: The bcs provided do not exist in the archive.
+        ValueError: The incorrect number of labels are provided.
+    """
+    # Try getting the colormap early in case it fails.
+    cmap = _retrieve_cmap(cmap)
+
+    #retrieve data from archive
+    archive_data = archive.as_pandas(include_solutions=False)
```

Comment:
It may be beneficial to sort the entries by objective, so that the brightest entries get plotted last. On the other hand, this would kill people who have a lot of entries in their archive. Maybe make it an option?